js/main.js

```diff
@@ -1,2 +1,95 @@
 /* global data */
 /* exported data */
+
+var urlInput = document.querySelector('#input-url');
+var photo = document.querySelector('#main-image');
+
+function updatePhoto(event) {
+  photo.setAttribute('src', event.target.value);
+  if (event.target.value === '') {
+    photo.setAttribute('src', 'images/placeholder-image-square.jpg');
+  }
+}
+
+urlInput.addEventListener('input', updatePhoto);
+
+var form = document.querySelector('#form');
+var title = document.querySelector('#title-text');
+var notes = document.querySelector('#notes-text');
+
+function submitListener(event) {
+  data.entries.unshift({
+    title: title.value,
+    photoURL: urlInput.value,
+    notes: notes.value,
+    entryID: data.nextEntryId
+  });
+  data.nextEntryId++;
+
+  entryFormContainer.className = 'container hidden';
+  entriesMainContainer.className = 'container';
+
+  photo.setAttribute('src', 'images/placeholder-image-square.jpg');
+  form.reset();
+}
+
+form.addEventListener('submit', submitListener);
+
+var entryUL = document.querySelector('#entry-list');
+var noEntriesMessage = document.querySelector('#no-entries');
+
+function entryAdd(entry) {
+  var entryListing = document.createElement('li');
+  var firstChild = entryUL.firstChild;
+  entryListing.className = 'row entry';
+  entryUL.insertBefore(entryListing, firstChild);
+
+  var entryImage = document.createElement('img');
+  entryImage.className = 'column-half';
+  entryImage.setAttribute('src', entry.photoURL);
+  entryListing.appendChild(entryImage);
+
+  var entryDiv = document.createElement('div');
+  entryDiv.className = 'column-half';
+  entryListing.appendChild(entryDiv);
+
+  var entryTitle = document.createElement('h2');
+  var entryTitleText = document.createTextNode(entry.title);
+  entryTitle.appendChild(entryTitleText);
+  entryDiv.appendChild(entryTitle);
+
+  var entryNotes = document.createElement('p');
+  var entryNotesText = document.createTextNode(entry.notes);
+  entryNotes.appendChild(entryNotesText);
+  entryDiv.appendChild(entryNotes);
+}
+
+window.addEventListener('DOMContentLoaded', domContentLoadedListener);
+
+function domContentLoadedListener(event) {
+  if (data.nextEntryId !== 1) {
```

Comment:
I would instead check if `data.entries` has no values inside of it. Once entries can be deleted, you might have a higher value for the next entry id, but no entries currently existing.